arch/arm/src/armv6-m/arm_doirq.c

```diff
@@ -84,6 +84,12 @@ uint32_t *arm_doirq(int irq, uint32_t *regs)
 
       irq_dispatch(irq, regs);
 #endif
+      if (tcb->sigdeliver)
+        {
+          irqstate_t flags = enter_critical_section();
```

Comment:
why need critical section

Reply:
removed critical_section and add comment.
/* Pendsv able to access running tcb with no critical section */